### ISSUE TITLE
<br> doubled in HTML labels and explanation control

### ISSUE THREAD

**Issue by acspike** [orbeon/orbeon-forms#2177]
Open the following form. Notice two lines broken with a single `<br />` tag. Click the **Insert Row Above** arrow icon. Notice that there are now two `<br />` tags separating the lines. Additional `<br />`s are also added when adding new controls to the grid or applying control settings for another control in the grid.

```xml
<xh:html xmlns:xh="http://www.w3.org/1999/xhtml" xmlns:xf="http://www.w3.org/2002/xforms"
         xmlns:xs="http://www.w3.org/2001/XMLSchema"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xmlns:ev="http://www.w3.org/2001/xml-events"
         xmlns:xi="http://www.w3.org/2001/XInclude"
         xmlns:xxi="http://orbeon.org/oxf/xml/xinclude"
         xmlns:xxf="http://orbeon.org/oxf/xml/xforms"
         xmlns:exf="http://www.exforms.org/exf/1-0"
         xmlns:fr="http://orbeon.org/oxf/xml/form-runner"
         xmlns:saxon="http://saxon.sf.net/"
         xmlns:sql="http://orbeon.org/oxf/xml/sql"
         xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/"
         xmlns:fb="http://orbeon.org/oxf/xml/form-builder">
    <xh:head>
        <xh:title>Test Form</xh:title>
        <xf:model id="fr-form-model" xxf:expose-xpath-types="true">

            <!-- Main instance -->
            <xf:instance id="fr-form-instance" xxf:exclude-result-prefixes="#all">
                <form>
                    <section-1>
                        <control-2/>
                        
                    </section-1>
                </form>
            </xf:instance>

            <!-- Bindings -->
            <xf:bind id="fr-form-binds" ref="instance('fr-form-instance')">
                <xf:bind id="section-1-bind" name="section-1" ref="section-1">
                    <xf:bind id="control-2-bind" ref="control-2" name="control-2"/>
                    
                </xf:bind>
            </xf:bind>

            <!-- Metadata -->
            <xf:instance xxf:readonly="true" id="fr-form-metadata" xxf:exclude-result-prefixes="#all">
                <metadata>
                    <application-name>Athletics</application-name>
                    <form-name>TestForm</form-name>
                    <title xml:lang="en">Test Form</title>
                    <description xml:lang="en"/>
                    <singleton>false</singleton>
                </metadata>
            </xf:instance>

            <!-- Attachments -->
            <xf:instance id="fr-form-attachments" xxf:exclude-result-prefixes="#all">
                <attachments>
                    <css mediatype="text/css" filename="" size=""/>
                    <pdf mediatype="application/pdf" filename="" size=""/>
                </attachments>
            </xf:instance>

            <!-- All form resources -->
            <!-- Don't make readonly by default in case a service modifies the resources -->
            <xf:instance id="fr-form-resources" xxf:readonly="false" xxf:exclude-result-prefixes="#all">
                <resources>
                    <resource xml:lang="en">
                        <control-2>
                            <text>&lt;div&gt;line1&lt;br /&gt;line2&lt;/div&gt;</text>
                        </control-2>
                        <section-1>
                            <label>Untitled Section</label>
                        </section-1>
                        
                    </resource>
                </resources>
            </xf:instance>

            <!-- Utility instances for services -->
            <xf:instance id="fr-service-request-instance" xxf:exclude-result-prefixes="#all">
                <request/>
            </xf:instance>

            <xf:instance id="fr-service-response-instance" xxf:exclude-result-prefixes="#all">
                <response/>
            </xf:instance>

        </xf:model>
    </xh:head>
    <xh:body>
        <fr:view>
            <fr:body xmlns:xbl="http://www.w3.org/ns/xbl"
                     xmlns:oxf="http://www.orbeon.com/oxf/processors"
                     xmlns:p="http://www.orbeon.com/oxf/pipeline">
                <fr:section id="section-1-control" bind="section-1-bind">
                    <xf:label ref="$form-resources/section-1/label"/>
                    <fr:grid>
                        <xh:tr>
                            
                            <xh:td>
                                <fr:explanation xmlns:xxbl="http://orbeon.org/oxf/xml/xbl"
                                                xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
                                                id="control-2-control"
                                                bind="control-2-bind">
                                    <xf:alert ref="$fr-resources/detail/labels/alert"/>
                                    <fr:text ref="$form-resources/control-2/text" mediatype="text/html"/>
                                </fr:explanation>
                            </xh:td>
                        </xh:tr>
                    </fr:grid>
                </fr:section>
            </fr:body>
        </fr:view>
    </xh:body>
</xh:html>
```

**Comment by acspike** [comment]
TagSoup doesn't seem to cause the doubling. Or at least, TagSoup doesn't do it in a simple case on the command line:

```
> echo -e "<html><head></head><body>test1<br/>test2</body></html>" | java -jar lib/tagsoup-1.2.jar --html | java -jar lib/tagsoup-1.2.jar --html
<html><head></head><body>test1<br>test2</body></html>
```

**Comment by acspike** [comment]
The xxf:event-response returned by the server includes a closing `</br>` tag. Which must be reinterpreted by the client as a second `<br>`. 

```
&lt;label class="xforms-label" for="fb≡section-1-control≡tmp-6-tmp≡control-1-control≡xforms-input-1"&gt;apples&lt;br&gt;&lt;/br&gt;oranges&lt;/label&gt;
```

**Comment by avernet** [comment]
@acspike We'll look into this after the 4.9 release.

**Comment by avernet** [comment]
This seems to happen when a full update is triggered (without which things are working fine). To reproduce, run the following, click on the button, and notice the `<br></br>` in the response, which is an improper HTML serialization of `<br>`, as in HTML5 `<br></br>` is equivalent to `<br><br>`.

```xml
<xh:html xmlns:xh="http://www.w3.org/1999/xhtml"
         xmlns:xf="http://www.w3.org/2002/xforms"
         xmlns:xxf="http://orbeon.org/oxf/xml/xforms"
         xmlns:ev="http://www.w3.org/2001/xml-events"
         xmlns:xs="http://www.w3.org/2001/XMLSchema"
         xmlns:fr="http://orbeon.org/oxf/xml/form-runner">
  <xh:head>
    <xf:model xxf:ajax.update.full.threshold="0">
      <xf:instance>
        <person>
          <first-name xmlns="">Gaga</first-name>
        </person>
      </xf:instance>
    </xf:model>
  </xh:head>
  <xh:body>
    <xf:group xxf:update="full">
      <xf:output id="br" ref="first-name" mediatype="text/html"/>
    </xf:group>
    <xf:trigger>
      <xf:label>Set to br</xf:label>
      <xf:setvalue ev:event="DOMActivate" ref="first-name" value="'&lt;br&gt;&lt;/br&gt;'"/>
    </xf:trigger>
  </xh:body>
</xh:html>
```

**Comment by ebruchez** [comment]
And this is because we use [`HTMLFragmentSerializer.java`](https://github.com/orbeon/orbeon-forms/blob/d494e07247221473638708b8c5b296c960095ce8/src/main/java/org/orbeon/oxf/xml/HTMLFragmentSerializer.java) to serialize to HTML.

This serializer was written specifically to handle full updates. But obviously it is broken.
